### DISCUSSION TITLE
Importing DAG files in Airflow 2.4.x

### DISCUSSION THREAD

**Discussion by Limess** [apache/airflow#27255] [Q&A]
We've started seeing the following issue when trying to upgrading to Airflow 2.4.x:

```
dags/signal_ai/data_platform/articles/offline_processing_articles_hudi.py: Ignoring DAG offline-processing-hudi-articles-from-realtime-v3 from dags/signal_ai/data_platform/articles/offline_processing_articles_hudi.py - also found in dags/signal_ai/data_platform/articles/offline_processing_articles_verify_deletion.py
```

The issue is that `dags/signal_ai/data_platform/articles/offline_processing_articles_verify_deletion.py` imports `dags/signal_ai/data_platform/articles/offline_processing_articles_hudi.py` in order to setup an ExternalTaskSensor on the correct DAG ID.

This is similar to this issue: https://github.com/apache/airflow/issues/26936 and is a result of this change https://airflow.apache.org/blog/airflow-2.4.0/#auto-register-dags-used-in-a-context-manager-no-more-as-dag-needed.

In general, we semi-frequently import other files containing DAGs - usually to get the correct `dag_id` to use in an ExternalTaskSensor, (to avoid having to update code in multiple changes when we change a DAG name to do psuedo-versioning). This wasn't an issue in previous Airflow versions.

This leads to two questions:
* Is there a better approach to avoid importing other DAG files, but still get the relevant DAG ID?
* Is there still a convenient way of importing files which create DAGs without failing to load the DagBag, without modifying the `auto_register` parameter?

**Comment by ephraimbuddy** [comment]
Another thing you can do is use `if __name__ == '__main__'` to control what happens when imported. For example, using the code in #26936 you can do something like this:
```
from datetime import datetime
from airflow.decorators import dag, task

class BaseDag:
    START_DATE = datetime(2022, 1, 1)

    def __init__(self, message: str):
        self.message = message

    def dag_wrapper(self, dag_id: str):
        @dag(dag_id=dag_id, start_date=self.START_DATE, catchup=False)
        def _base_dag():

            @task(task_id="print_message")
            def print_message(message: str):
                print(message)

            print_message(self.message)

        return _base_dag()

if __name__ == '__main__':
    BaseDag("my message").dag_wrapper("BaseDag")
```
Maybe not a better approach
